bpf/bpf_host.c

```diff
@@ -854,7 +858,11 @@ handle_ipv4_cont(struct __ctx_buff *ctx, __u32 secctx, const bool from_host,
 	if (from_proxy && info->tunnel_endpoint && encrypt_key)
 		return set_ipsec_encrypt(ctx, encrypt_key, info->tunnel_endpoint,
 					 info->sec_identity, true, false);
-#endif
+
+	if (from_proxy &&
+	    (!info || !identity_is_cluster(info->sec_identity)))
+		ctx->mark = MARK_MAGIC_PROXY_TO_WORLD;
+#endif /* ENABLE_IPSEC && !TUNNEL_MODE */
```

Comment:
@jschwinger233 coming back to this - I only now realized that https://github.com/cilium/cilium/issues/36329 also applies here, and the mark potentially gets scrubbed on the transfer between `cilium_host` and `cilium_net`. Which is why we're still setting `use_meta=true` in the `set_ipsec_encrypt()` call above.

So I suppose this mark works well as mechanism to skip the transparent-socket match in iptables - but we can't really trust yet that it also survives until `cilium_net` (or a subsequent `to-netdev`) would see it.

Reply:
:fearful: